src/main/java/org/truffleruby/Layouts.java

```diff
@@ -28,6 +28,6 @@
 
     // Frame slot name for special variable storage.
 
-    public static final String SPECIAL_VARIABLES_STORAGE = TranslatorEnvironment.TEMP_PREFIX + "$~_";
+    public static final String SPECIAL_VARIABLES_STORAGE = "%$~_";
```

Comment:
I think this is quite unclear, and the constant has the big advantage to clarify what is the prefix for (to hide the variable).
Could you remove this commit?

Reply:
The whole thing of the separate string for the prefix? Seems so silly to spend startup time creating dynamic strings and interning them when they're so simple.

Reply:
javac actually folds such a simple expression, so no startup cost:
```
$ javap -v ./mxbuild/jdk11/org.truffleruby/bin/org/truffleruby/Layouts.class
...
  public static final java.lang.String SPECIAL_VARIABLES_STORAGE;
    descriptor: Ljava/lang/String;
    flags: (0x0019) ACC_PUBLIC, ACC_STATIC, ACC_FINAL
    ConstantValue: String %$~_
...
```
(on master, not this branch)

My issue with this is it's explained nowhere what `%` is for, and that can make it rather confusing for people not already knowing it.

Reply:
The rest of the code does it in a method though - and then calls `intern`.

Shall we comment it? Is that a good halfway?

Reply:
The `prefixName` helper is not needed, so I'd say let's just inline, and could also be without the `intern()` since I'd think javac folding it means interning it too (and we have asserts in place that all variables are interned).